cli/daemon/namespace/namespace.go

```diff
@@ -0,0 +1,327 @@
+	}
+	defer rows.Close()
+	var nss []*Namespace
+
+	for rows.Next() {
+		var ns Namespace
+		if err := rows.Scan(&ns.ID, &ns.Name, &ns.Active, &ns.CreatedAt, &ns.LastActiveAt); err != nil {
+			return nil, errors.Wrap(err, "scan namespace")
+		}
+		ns.App = app
+		nss = append(nss, &ns)
+	}
+	if err := rows.Err(); err != nil {
+		return nil, errors.Wrap(err, "list namespaces")
+	}
+
+	// If we have no namespaces at all, create a default one.
+	if len(nss) == 0 {
+		ns, err := m.Create(ctx, app, "default")
+		if err != nil {
+			return nil, err
+		}
+		nss = []*Namespace{ns}
+	}
+
+	return nss, nil
+}
+
+func (m *Manager) GetByName(ctx context.Context, app *apps.Instance, name Name) (*Namespace, error) {
+	var ns Namespace
+	err := m.db.QueryRowContext(ctx, `
+		SELECT id, name, active, created_at, last_active_at
+		FROM namespace
+		WHERE app_id = ? AND name = ?
+	`, app.PlatformOrLocalID(), name).Scan(&ns.ID, &ns.Name, &ns.Active, &ns.CreatedAt, &ns.LastActiveAt)
+	if err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			return nil, ErrNotFound
+		}
+		return nil, errors.Wrap(err, "get namespace")
+	}
+	ns.App = app
+	return &ns, nil
+}
+
+func (m *Manager) GetByID(ctx context.Context, app *apps.Instance, id ID) (*Namespace, error) {
+	var ns Namespace
+	err := m.db.QueryRowContext(ctx, `
+		SELECT id, name, active, created_at, last_active_at
+		FROM namespace
+		WHERE app_id = ? AND id = ?
+	`, app.PlatformOrLocalID(), id).Scan(&ns.ID, &ns.Name, &ns.Active, &ns.CreatedAt, &ns.LastActiveAt)
+	if err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			return nil, ErrNotFound
+		}
+		return nil, errors.Wrap(err, "get namespace")
+	}
+	ns.App = app
+	return &ns, nil
+}
+
+func (m *Manager) Delete(ctx context.Context, app *apps.Instance, name Name) error {
+	tx, err := m.db.BeginTx(ctx, nil)
+	if err != nil {
+		return err
+	}
+	defer tx.Rollback() // committed explicitly on success
+
+	var ns Namespace
+	err = tx.QueryRowContext(ctx, `
+		DELETE FROM namespace
+		WHERE app_id = ? AND name = ?
+		RETURNING id, name, active, created_at, last_active_at
+	`, app.PlatformOrLocalID(), name).Scan(&ns.ID, &ns.Name, &ns.Active, &ns.CreatedAt, &ns.LastActiveAt)
+	if ns.Active {
+		return ErrActive
+	}
+
+	// Check all the deletion handlers.
+	for _, h := range m.handlers {
+		if err := h.CanDeleteNamespace(ctx, app, &ns); err != nil {
+			return errors.Newf("cannot delete namespace: %v", err)
+		}
+	}
+
+	// Actually delete the namespace.
+	for _, h := range m.handlers {
+		if err := h.DeleteNamespace(ctx, app, &ns); err != nil {
+			return errors.Newf("failed to delete namespace: %v", err)
+		}
+	}
+
+	err = tx.Commit()
+	return errors.Wrap(err, "delete namespace")
+}
+
+func (m *Manager) Switch(ctx context.Context, app *apps.Instance, name Name) (*Namespace, error) {
+	// Resolve the namespace to switch to.
+	var target *Namespace
+
+	// If the name is "-", switch to the previous namespace.
+	if name == "-" {
```

Comment:
I didn't see this documented on the commands or else where in the code; just deep in the manager here.

We should add this to the command docs